### PR TITLE
feat(homepage): recent insights list

### PR DESCRIPTION
**Pull request by rcmarron**
## Problem

Right now, there's no way to get back to a recently viewed insight. This is a common thing users want to do, so we should make it as easy as possible.

## Changes

Behind the FF `homepage-lists`, this PR adds a list to the homepage that shows the user's recent insights.

[Figma designs here](https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=7434%3A38795)

https://user-images.githubusercontent.com/4813045/163277202-2845d9a7-878d-435c-ab4c-85fee8f7c2f7.mov

Empty list view:
image

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Added tests to the API for recent insights

### CODE REVIEW

File: frontend/src/scenes/project-homepage/ProjectHomepage.tsx [hunk 1]
```diff
@@ -1,71 +1,43 @@
...
-        sessionRecordingsTableLogicInstance
-    )
-    const { closeSessionPlayer } = useActions(sessionRecordingsTableLogicInstance)
     const { featureFlags } = useValues(featureFlagLogic)
     const { currentTeam } = useValues(teamLogic)
     const { dashboard } = useValues(dashboardLogic)
     const { showInviteModal } = useActions(inviteLogic)
     const { showPrimaryDashboardModal } = useActions(primaryDashboardModalLogic)
 
+    const [topListContainerWidth, setTopListContainerWidth] = useState<number>()
```
Comment (alexkim205) [inline frontend/src/scenes/project-homepage/ProjectHomepage.tsx:33]: Could we use useSize here to only keep track of the width of the container?

File: frontend/src/scenes/insights/insightLogic.ts [hunk 2]
```diff
@@ -562,6 +563,17 @@ export const insightLogic = kea<insightLogicType>({
...
+            // Report the insight being viewed to our '/viewed' endpoint. Used for "recently viewed insights"
+
+            // NOTE: This should be merged into the same action as `reportInsightViewed`, but we can't right now
+            // because there are some issues with `reportInsightViewed` not being called when the
+            // insightLogic is already loaded.
+            // For example, if the user navigates to an insight after viewing it on a dashboard, `reportInsightViewed`
+            // will not be called. This should be fixed when we refactor insightLogic, but the logic is a bit tangled
+            // right now
+            api.create(`api/projects/${teamLogic.values.currentTeamId}/insights/${values.insight.id}/viewed`)
+        },
```
Comment (alexkim205) [inline frontend/src/scenes/insights/insightLogic.ts:576]: I think I understand what you're trying to do here, but perhaps a TODO here could help signal that this should be removed once the existing `reportInsightViewed` bug is fixed.

File: frontend/src/scenes/project-homepage/recentInsightsLogic.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    connect: {
+        values: [teamLogic, ['currentTeamId']],
+    },
+    loaders: ({ values }) => ({
+        recentInsights: [
+            [] as InsightModel[],
+            {
+                loadRecentInsights: async () => {
+                    const response = await api.get(
+                        `api/projects/${values.currentTeamId}/insights/?my_last_viewed=true&order=-my_last_viewed_at`
```
Comment (alexkim205) [inline frontend/src/scenes/project-homepage/recentInsightsLogic.ts:19]: I think we're moving away from using the currentTeamId pattern and towards the api builder which handles that for us. Could we create another entry in `frontend/src/lib/api.ts` for `insights`?

File: frontend/src/scenes/project-homepage/recentInsightsLogic.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                    )
+                    return response.results
+                },
+            },
+        ],
+    }),
+
+    events: ({ actions }) => ({
+        afterMount: () => {
+            actions.loadRecentInsights()
```
Comment (alexkim205) [inline frontend/src/scenes/project-homepage/recentInsightsLogic.ts:29]: If it's not too much of a hassle, it'd be icing on the cake if we have a kea logic test for this.

File: frontend/src/scenes/saved-insights/SavedInsights.tsx [hunk 5]
```diff
@@ -91,6 +91,14 @@ export const scene: SceneExport = {
     logic: savedInsightsLogic,
 }
 
+export function InsightIcon({ insight }: { insight: InsightModel }): JSX.Element | null {
```
Comment (alexkim205) [inline frontend/src/scenes/saved-insights/SavedInsights.tsx:94]: nice refactor :)

**Review by alexkim205 — DISMISSED** [review]
Looking real nice 💯! Just left a some code comments in line for your suggestions

**Review by clarkus — APPROVED** [review]
Looks good to me - the one issue I see is that we're labeling the timestamp for recently viewed insights (last modified) but we don't label the time value for recordings. It's not a huge deal, but it would be good to clarify that date value with a label of some kind.

**Review by clarkus — COMMENTED** [review]
Should "your projects" be possessive in this context? The button label is basically the same as the tertiary  link in the header "view all". It feels like the button should have more active language - "view saved insights" or "view insights" - I'm not sure we need to specify "saved insights" because all the insights are saved in the current model.

File: frontend/src/scenes/project-homepage/ProjectHomepage.tsx (hunk 1 above)
Comment (rcmarron) [reply frontend/src/scenes/project-homepage/ProjectHomepage.tsx]: Ooooo nice! didn't know that was a thing!

File: frontend/src/scenes/project-homepage/recentInsightsLogic.ts (hunk 3 above)
Comment (rcmarron) [reply frontend/src/scenes/project-homepage/recentInsightsLogic.ts]: Didn't know this new pattern was a thing! I like it.

I took a look at moving `/insights` into it, and it looks like a pretty good refactor as we use the endpoint all over. Thoughts on it being a separate PR?

File: frontend/src/scenes/project-homepage/recentInsightsLogic.ts (hunk 4 above)
Comment (rcmarron) [reply frontend/src/scenes/project-homepage/recentInsightsLogic.ts]: Just moved this into the `projectHomepageLogic` and added a test!

**Comment by rcmarron** [comment]
@clarkus Good suggestions! I just cleaned up the copy to be:
image

File: frontend/src/scenes/project-homepage/recentInsightsLogic.ts (hunk 3 above)
Comment (alexkim205) [reply frontend/src/scenes/project-homepage/recentInsightsLogic.ts]: That sounds totally good to me! 👍

**Review by alexkim205 — APPROVED** [review]
:shipit:

**Review by clarkus — APPROVED** [review]

**Pull request merged**